backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/ConvertDiskCommand.java

```diff
@@ -167,11 +167,22 @@ protected void executeCommand() {
             return;
         }
 
-        long requiredSize =  actionReturnValue.getActionReturnValue();
+        long requiredSize = actionReturnValue.getActionReturnValue();
+
+        CreateVolumeContainerCommandParameters parameters = createVolumeCreationParameters(getDiskImage(), requiredSize);
+
+        // Initial size will not be used for this configuration, meaning the required size will not
+        // be used at all. However, when converting to COW/Preallocated the required size may be larger
+        if (parameters.getVolumeType() == VolumeType.Preallocated &&
+                parameters.getVolumeFormat() == VolumeFormat.COW) {
+            parameters.setSize(Math.max(getDiskImage().getSize(), requiredSize));
+            if (parameters.getSize() > getDiskImage().getSize()) {
+                log.info("Updated disk's size to: '{}'", parameters.getSize());
```

Comment:
Isn't that better to have a auditl log message here in AuditLogMessages.properties and log also the disk name ?

Reply:
I can add the disk's id, but I don't think it's audit log material as this is an internal implementation detail.

Audit logs are generally missing in this flow and can be added separately

Reply:
disk ID was added